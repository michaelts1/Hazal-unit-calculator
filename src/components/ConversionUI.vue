<script>
import ConversionUIUnitTable from './ConversionUIUnitTable.vue'
import RabiSelect from './RabiSelect.vue'

export default {
	components: {
		RabiSelect,
		ConversionUIUnitTable,
	},

	props: {
		/* eslint-disable-next-line */ // If this useless line is removed, Vetur breaks :shrug:
		uselessProp: undefined,
	},

	data() {
		return {
			ruler: 'גרח',
		}
	},

	computed: {
		etzba() { // Base unit is meter
			switch (this.ruler) {
				case 'חזון_איש':
					return 0.024
				case 'גרח':
				default:
					return 0.02
			}
		},
		tefach: self => 4    * self.etzba ,
		sit   : self => 2    * self.tefach,
		zeret : self => 0.5  * self.amah  ,
		amah  : self => 6    * self.tefach,
		ris   : self => 2/15 * self.mil   ,
		mil   : self => 2000 * self.amah  ,
		parsa : self => 4    * self.mil   ,
		units() {
			return [
				{ hidden: true, name: 'סנטימטר', value: 0.01 },
				{ hidden: true, name: 'מטר', value: 1 },
				{ hidden: true, name: 'קילומטר', value: 1000 },
				{ hidden: false, name: 'אצבע', value: this.etzba },
				{ hidden: false, name: 'טפח', value: this.tefach },
				{ hidden: false, name: 'סיט', value: this.sit },
				{ hidden: false, name: 'זרת', value: this.zeret },
				{ hidden: false, name: 'אמה', value: this.amah },
				{ hidden: false, name: 'ריס', value: this.ris },
				{ hidden: false, name: 'מיל', value: this.mil },
				{ hidden: false, name: 'פרסה', value: this.parsa },
			]
		},
	},
}
</script>

<template>
	<RabiSelect
		:default-ruler="ruler"
		@ruler-change="newRuler => ruler = newRuler"
	/>

	<h6>שים לב: הסיט מחושב לפי שיטת הרמב"ם</h6>
	<ConversionUIUnitTable :units="units" />
</template>
