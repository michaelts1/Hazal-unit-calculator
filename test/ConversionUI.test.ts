import { describe, expect, it } from 'vitest'
import ConversionUI from '../src/components/ConversionUI.vue'
import { mount } from '@vue/test-utils'

describe('Component ConversionUI', () => {
	it('exists', () => {
		expect(ConversionUI).toBeTruthy()
	})

	const wrapper = mount(ConversionUI)

	it('matches snapshot', () => {
		expect(wrapper.html()).toMatchSnapshot()
	})

	it('responds to events emitted by RabiSelect', async () => {
		const RabiSelect = wrapper.findComponent({ name: 'RabiSelect' })

		expect(wrapper.vm.etzba).toBe(0.02) // Default
		RabiSelect.vm.$emit('ruler-change', 'חזון_איש')
		expect(wrapper.vm.etzba).toBe(0.024) // New value
	})
})
