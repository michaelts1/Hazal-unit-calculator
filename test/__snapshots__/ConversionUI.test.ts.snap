// Vitest Snapshot v1

exports[`Component ConversionUI > matches snapshot 1`] = `
"<label for=\\"ruler-select\\">שיטת מדידה: </label><select id=\\"ruler-select\\">
  <option value=\\"גרח\\"> גר\\"ח (רב חיים נאה) </option>
  <option value=\\"חזון_איש\\"> חזון איש </option>
</select>
<h6>שים לב: הסיט מחושב לפי שיטת הרמב\\"ם</h6>
<table data-v-6edac4c0=\\"\\">
  <thead data-v-6edac4c0=\\"\\">
    <tr data-v-6edac4c0=\\"\\">
      <th data-v-6edac4c0=\\"\\">אצבע</th>
      <th data-v-6edac4c0=\\"\\">טפח</th>
      <th data-v-6edac4c0=\\"\\">סיט</th>
      <th data-v-6edac4c0=\\"\\">זרת</th>
      <th data-v-6edac4c0=\\"\\">אמה</th>
      <th data-v-6edac4c0=\\"\\">ריס</th>
      <th data-v-6edac4c0=\\"\\">מיל</th>
      <th data-v-6edac4c0=\\"\\">פרסה</th>
    </tr>
  </thead>
  <tbody data-v-6edac4c0=\\"\\">
    <tr data-v-6edac4c0=\\"\\">
      <td data-v-6edac4c0=\\"\\">2 ס\\"מ</td>
      <td data-v-6edac4c0=\\"\\">8 ס\\"מ</td>
      <td data-v-6edac4c0=\\"\\">16 ס\\"מ</td>
      <td data-v-6edac4c0=\\"\\">24 ס\\"מ</td>
      <td data-v-6edac4c0=\\"\\">48 ס\\"מ</td>
      <td data-v-6edac4c0=\\"\\">128 מטר</td>
      <td data-v-6edac4c0=\\"\\">960 מטר</td>
      <td data-v-6edac4c0=\\"\\">3.84 ק\\"מ</td>
    </tr>
  </tbody>
</table>
<div class=\\"wrapper-row\\" data-v-6edac4c0=\\"\\">
  <div class=\\"wrapper-column\\" data-v-6edac4c0=\\"\\"><select class=\\"centered\\" data-v-6edac4c0=\\"\\">
      <option value=\\"סנטימטר\\">סנטימטר</option>
      <option value=\\"מטר\\">מטר</option>
      <option value=\\"קילומטר\\">קילומטר</option>
      <option value=\\"אצבע\\">אצבע</option>
      <option value=\\"טפח\\">טפח</option>
      <option value=\\"סיט\\">סיט</option>
      <option value=\\"זרת\\">זרת</option>
      <option value=\\"אמה\\">אמה</option>
      <option value=\\"ריס\\">ריס</option>
      <option value=\\"מיל\\">מיל</option>
      <option value=\\"פרסה\\">פרסה</option>
    </select><input class=\\"conversion-input\\" size=\\"12\\" data-v-951c0080=\\"\\" data-v-6edac4c0=\\"\\"></div>
  <div class=\\"wrapper-column\\" data-v-6edac4c0=\\"\\"> = </div>
  <div class=\\"wrapper-column\\" data-v-6edac4c0=\\"\\"><select class=\\"centered\\" data-v-6edac4c0=\\"\\">
      <option value=\\"סנטימטר\\">סנטימטר</option>
      <option value=\\"מטר\\">מטר</option>
      <option value=\\"קילומטר\\">קילומטר</option>
      <option value=\\"אצבע\\">אצבע</option>
      <option value=\\"טפח\\">טפח</option>
      <option value=\\"סיט\\">סיט</option>
      <option value=\\"זרת\\">זרת</option>
      <option value=\\"אמה\\">אמה</option>
      <option value=\\"ריס\\">ריס</option>
      <option value=\\"מיל\\">מיל</option>
      <option value=\\"פרסה\\">פרסה</option>
    </select><input class=\\"conversion-input\\" disabled=\\"\\" size=\\"12\\" data-v-951c0080=\\"\\" data-v-6edac4c0=\\"\\"></div>
</div>"
`;
